### DISCUSSION TITLE
Why data2vec is not popular compared to HuBERT or WavLM?

### DISCUSSION THREAD

**Discussion by raotnameh** [s3prl/s3prl#553] [General]
data2vec is clearly a better model for the ASR task, then why is not used extensively? This holds true for the base model. For large models there is no clear winner except in very low resource scenarios. Table 2 in the original [dat2vec](https://arxiv.org/pdf/2202.03555) paper

**Comment by leo19941227** [comment]
I have the same question, but I don't know the answer. The only answer I use to convince myself is that HuBERT is conceptually simpler to implement compared to the EMA in data2vec since it is just the classification. Similarly, people tend to use BEST-RQ since the random quantizer further remove the reliance on the offline clustering procedure and make the whole pipeline super clean. So my two cents on this question is that when the performance difference is not super significant, simplicity is more important.

**Reply by raotnameh** [reply]
Which would be a better choice, discrete tokens ids extracted from HuBERT or data2vec?

**Reply by leo19941227** [reply]
If you want the discrete tokens to represent only the content information, then I would suggest data2vec. I would further suggest to use some models with special techniques which remove the speaker variations like [spin](https://github.com/vectominist/spin)

**Reply by raotnameh** [reply]
HuBERT was trained to predict discrete tokens explicitly. Don't you think, HuBERT would have an edge if simply using clustering techniques to extract discrete ids?

**Reply by raotnameh** [reply]
I think, we would have to wait for the someone to do a comparative analysis similar to figure 4 in the paper [COMPARATIVE LAYER-WISE ANALYSIS OF SELF-SUPERVISED SPEECH MODELS](https://arxiv.org/pdf/2211.03929).

**Comment by raotnameh** [comment]
From what I got after talking to other people over email is: 

1. The difference between WavLM and data2vec in the ASR task is marginal, but overall, WavLM is better when considering all other tasks than ASR. And we don’t care about small differences in pre-training. When using the LARGE models. 

2. Data2vec seems to be used less in general and the code for data2vec is also a bit less easy to use.

**Reply by raotnameh** [reply]
On the SUPERB benchmark: 
WavLM Large: ASR > 3.44 and PR > 3.06
data2vec Large: ASR > 3.36 and PR > 3.6 

WavLM : ASR > 4.94 and PR > 4.69
data2vec : ASR > 6.21 and PR > 4.84 
MS-HuBERT: ASR > 5.32 and PR > 4.17.

**Reply by raotnameh** [reply]
Now, I see why data2vec isn't popular, as most people rely on large models. The core difference is in the base case, where the ASR and PR difference is significant, but no one is using them. The base case difference can be alleviated with a simple change in the loss function implementation from this paper : [MS-HuBERT](https://arxiv.org/pdf/2406.05661)
